lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -133,6 +133,23 @@ namespace {
     ///     live-in values used within the loop (with no SILArgument specified).
     SmallVector<std::pair<SILArgument*, TF_Output>, 4> outputs;
 
+    /// The tentative outputs that may be "upgraded" to `outputs`, if the graph
+    /// function we are lowering ends with a conditional region (case #2
+    /// above). One example is an outer if whose true (or false) body consists
+    /// solely of another conditional region (an inner if). In that case,
+    /// `outputs` is not filled in after the graph function of the inner if is
+    /// completed. To handle this case, when lowering the conditional region, we
+    /// always set the `tentative_outputs` of the graph function, so that we can
+    /// upgrade them to `outputs` in the above case when needed. On the other
+    /// case, the `tentative_outputs` is simply discarded, if after lowering the
+    /// conditional region, the graph function construction moves on to lower a
+    /// subsequent region.
+    ///
+    /// Invariant: At the completion of constructing a graph function, only one
+    /// of `output` and `tentative_outputs` should be set -- the former is set
+    /// in cases #1, #3, #4 above. The latter is set in case #2 described above.
+    SmallVector<std::pair<SILArgument *, TF_Output>, 4> tentative_outputs;
```

Comment:
I don't really understand the purpose of tentative_outputs, but it would be preferable to not add this to GraphFunctionBody, which is global.  Is there a way to make this local to the control flow lowering code?

Reply:
This is a temporary solution, and I believe it must be added to `GraphFunctionBody`, so that the subsequent region lowering logic in the same graph function can read/write the it.

@bgogul and/or I discussed a potentially better fix, by extending the definition of a conditional SESE region to have a merging block. We can discuss more on Monday.